### PR TITLE
doc: additional DM parameter documentation

### PR DESCRIPTION
**Pull request by dbkinder**
Update the device model documentation with a new user guide describing acrn-dm parameters 

Tracked-On: #2021

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 1]
```diff
@@ -90,6 +90,229 @@ options:
        --intr_monitor: enable interrupt storm monitor, params:
        		threshold/s,probe-period(s),delay_time(ms),delay_duration(ms)
 
+acrn-dm parameters
+==================
+
+Here are further descriptions for each of these parameters:
+
+A, acpi
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]: Can you add the `-` and `--` in front of each parameter? E.g. `-A, --acpi`

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 2]
```diff
@@ -90,6 +90,229 @@ options:
...
+   When assigning a PCI device as a passthrough device, we will reset it
+   first to get it to a valid device state. So if the device doesn't have
+   the reset capability, the passthrough will fail. The PCI device reset
+   can be disabled using this option.
+
+k, kernel <kernel_image_path>
+   Set the kernel (full path) for the UOS kernel. The maximum length is
+   1023. The DM handles bzImage image format.
+
+   usage: -k /path/to/your/kernel_image
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   usage: ``-k /path/to/your/kernel_image``
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 3]
```diff
@@ -90,6 +90,229 @@ options:
...
+
+r, ramdisk <ramdisk_image_path>
+   Set the ramdisk (full path) for the UOS. The maximum length is 1023.
+   The supported ramdisk format depends on your UOS kernel configuration.
+
+   usage: -r /path/to/your/ramdisk_image
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   usage: ``-r /path/to/your/ramdisk_image``
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 4]
```diff
@@ -90,6 +90,229 @@ options:
...
+
+B, bootargs <bootargs>
+   Set the UOS kernel command-line arguments. The maximum length is 1023.
+   The bootargs string will be passed to the kernel as its cmdline.
+
+   Example: -B "loglevel=7": it specifies the kernel log level at 7
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   Example: ``-B "loglevel=7"``: it specifies the kernel log level at 7
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 5]
```diff
@@ -90,6 +90,229 @@ options:
...
+   more than the total number of CPUs in the system, minus one (which is
+   used by the SOS).
+
+p, pincpu <vcpu:hostcpu>
+   Pin host CPU to appointed vCPU:
+
+   - vcpu is the ID of the CPU seen by the UOS, and
+   - hostcpu is the physical CPU ID on the system.
+
+   Example: -p  "1:2" means pin the 2nd physical cpu to 1st vcpu in UOS
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   Example: ``-p  "1:2"`` means pin the 2nd physical cpu to 1st vcpu in UOS
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 6]
```diff
@@ -90,6 +90,229 @@ options:
...
+   Setup PCI device configuration
+
+   slot_config format is::
+
+      <bus>:<slot>:<func>,<emul>[,<config>]
+      <slot>[:<func>],<emul>[,<config>]
+
+   Where:
+
+   - slot is 0..31
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   - ``slot`` is 0..31
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 7]
```diff
@@ -90,6 +90,229 @@ options:
...
+   - func is 0..7
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   - ``func`` is 0..7
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 8]
```diff
@@ -90,6 +90,229 @@ options:
...
+   - emul is a string describing the type of PCI device e.g. virtio-net
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   - ``emul`` is a string describing the type of PCI device e.g. virtio-net
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 9]
```diff
@@ -90,6 +90,229 @@ options:
...
+   - config is an optional device-dependent string, used for
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   - ``config`` is an optional device-dependent string, used for
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 10]
```diff
@@ -90,6 +90,229 @@ options:
...
+   This add virtual block in PCI slot 9 and use "/root/test.img" as the
+   disk image
+
+m, memsize <memory_size>
+   Setup total memory size for UOS
+
+   memory_size format is: "<size>{K/k, B/b, M/m, G/g}", and size is an
+   integer.
+
+   usage: -m 4g: set uos memory to 4 gigabytes.
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   usage: ``-m 4g``: set UOS memory to 4 gigabytes.
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 11]
```diff
@@ -90,6 +90,229 @@ options:
...
+   IOC DM opens ``/dev/ptmx`` device to create a peer PTY devices,  IOC DM uses
+   these to communicate with UART DM since UART DM needs a TTY capable
+   device as its backend.
+
+   The device model configuration command syntax for IOC mediator is::
+
+      -i,[ioc_channel_path],[wakeup_reason]
+      -l,[lpc_port],[ioc_channel_path]
+
+   - "ioc_channel_path" is an absolute path for communication between IOC
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   - ``ioc_channel_path`` is an absolute path for communication between IOC
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 12]
```diff
@@ -90,6 +90,229 @@ options:
...
+     mediator and UART DM.
+   - "lpc_port" is com1 or com2. IOC mediator needs one unassigned lpc
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   - ``lpc_port`` is com1 or com2. IOC mediator needs one unassigned lpc
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 13]
```diff
@@ -90,6 +90,229 @@ options:
...
+     port for data transfer between User OS and Service OS.
+   - "wakeup_reason" is IOC mediator boot up reason, where each bit represents
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   - ``wakeup_reason`` is IOC mediator boot up reason, where each bit represents
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 14]
```diff
@@ -90,6 +90,229 @@ options:
...
+     one wakeup reason.
+
+     Currently the wakeup reason bits supported by IOC firmware are:
+
+     - CBC_WK_RSN_BTN(bit 5): ignition button.
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
     - ``CBC_WK_RSN_BTN``(bit 5): ignition button.
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 15]
```diff
@@ -90,6 +90,229 @@ options:
...
+     - CBC_WK_RSN_RTC(bit 9): RTC timer.
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
     - ``CBC_WK_RSN_RTC``(bit 9): RTC timer.
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 16]
```diff
@@ -90,6 +90,229 @@ options:
...
+     - CBC_WK_RSN_DOR(bit 11): Car door.
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
     - ``CBC_WK_RSN_DOR``(bit 11): Car door.
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 17]
```diff
@@ -90,6 +90,229 @@ options:
...
+     - CBC_WK_RSN_SOC(bit 23): SoC active/inactive.
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
     - ``CBC_WK_RSN_SOC``(bit 23): SoC active/inactive.
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 18]
```diff
@@ -90,6 +90,229 @@ options:
...
+
+      --vsbl /usr/share/acrn/bios/VSBL.bin
+
+   uses ``/usr/share/acrn/bios/VSBL.bin`` as the vSBL image
+
+G, gvtargs <GVT_args>
+   ACRN implements GVT-g for graphics virtualization (aka AcrnGT). This
+   option allows you to set some of its parameters.
+
+   GVT_args format: "gvt_high_gm_sz gvt_low_gm_sz gvt_fence_sz"
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   GVT_args format: ``gvt_high_gm_sz gvt_low_gm_sz gvt_fence_sz``
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 19]
```diff
@@ -90,6 +90,229 @@ options:
...
+
+   Where:
+
+   - gvt_high_gm_sz: GVT-g aperture size, unit is MB
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   - ``gvt_high_gm_sz``: GVT-g aperture size, unit is MB
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 20]
```diff
@@ -90,6 +90,229 @@ options:
...
+   - gvt_low_gm_sz: GVT-g hidden gfx memory size, unit is MB
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   - ``gvt_low_gm_sz``: GVT-g hidden gfx memory size, unit is MB
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 21]
```diff
@@ -90,6 +90,229 @@ options:
...
+   - gvt_fence_sz: the number of fence registers
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]:
```suggestion
   - ``gvt_fence_sz``: the number of fence registers
```

**Comment by dbkinder** [comment]
@gvancuts OK.  I included your suggested edits, and moved this as a new user guide doc.

File: doc/user-guides/acrn-dm-parameters.rst [hunk 22]
```diff
@@ -0,0 +1,227 @@
...
+   default value.
+
+:kbd:`-B, --bootargs <bootargs>`
+   Set the UOS kernel command line arguments.
+   The maximum length is 1023.
+   The bootargs string will be passed to the kernel as its cmdline.
+
+   Example::
+
+      ``-B "loglevel=7"``
```
Comment (gvancuts) [inline doc/user-guides/acrn-dm-parameters.rst]:
```suggestion
      -B "loglevel=7"
```

File: doc/developer-guides/hld/hld-devicemodel.rst [hunk 23]
```diff
@@ -46,7 +46,7 @@ ISA, and PCI emulation.
 Configuration
 *************
 
-The acrn-dm should run according to below command line configuration
+The acrn-dm runs using these command line configuration
 options:
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-devicemodel.rst]: Can you also add a link to the manual after the `acrn-dm -v` output?

**Pull request merged**